src/Agent.Worker/Build/TrackingManager.cs

```diff
@@ -174,8 +174,17 @@ public sealed class TrackingManager : AgentService, ITrackingManager
             Trace.Entering();
 
             ArgUtil.NotNull(executionContext, nameof(executionContext));
-            string trackingFileLocation = GetTrackingFileLocation(executionContext);
-            return LoadIfExists(executionContext, trackingFileLocation);
+            // First, attempt to load the file from the old location (just collection and definition id)
+            string trackingFileLocation = GetTrackingFileLocation(executionContext, false);
+            var trackingConfig = LoadIfExists(executionContext, trackingFileLocation);
+            if (trackingConfig == null)
+            {
+                // If it's not in the old location, look for it in the new location
+                trackingFileLocation = GetTrackingFileLocation(executionContext, true);
+                trackingConfig = LoadIfExists(executionContext, trackingFileLocation);
+            }
```

Comment:
This is pretty nitty, but should we prefer the new location over the old location? That way once you're on the new configuration you don't technically need to reserve the old location (plus it saves the lookup)

Reply:
good points, it's been a while since I tested this so I will have to try that and see if there was a reason I chose to do it this way.

Reply:
Great suggestion! this is done.